### PR TITLE
Update Data docs based on feedback

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-INSERT_PR_#_HERE--keen-mayer-a86c8b.netlify.app/)

#### What does this PR do?

Some light touch follow ons based on feedback to:
- Refine wording to be less developer centric
- Show code snippets in Toolbar to convey these controls are intended to be in Toolbar components majority of time.
- Add cross links/related content
- 
#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

### CODE REVIEW

File: aries-site/src/pages/components/data.mdx [hunk 1]
```diff
@@ -12,10 +12,10 @@ import {
   ToolbarExample,
 } from '../../examples/components/';
 
-Data is the wrapper component and orchestrator for a set of subcomponents. Because this set of components work together to create the entirety of the experience, we like to refer to them as "Data and friends".
+Because this set of components work together to create the entirety of the experience, we like to refer to them as "Data and friends".
```
Comment (britt6612) [inline aries-site/src/pages/components/data.mdx:15]: I feel like Im asking myself "what set of components" while reading this

Comment (taysea) [reply aries-site/src/pages/components/data.mdx:15]: Should I maybe keep the first sentence, but just say "Data is the orchestrator for a set of subcomponents." (just cut out the "wrapper" bit)

Comment (jcfilben) [reply aries-site/src/pages/components/data.mdx:15]: ^ I agree the beginning also threw me off, felt like it needed a little more of an intro

Comment (taysea) [reply aries-site/src/pages/components/data.mdx]: Updated this

**Review by jcfilben — COMMENTED** [review]
Maybe unrelated to this PR but when I filter in the first example on the Data page the pagination seems incorrect. I would expect the pagination controls to not show and the summary should say 'Showing 1-6 of 21 items'

Screen Shot 2024-04-02 at 1 50 30 PM

**Comment by taysea** [comment]
> Maybe unrelated to this PR but when I filter in the first example on the Data page the pagination seems incorrect. I would expect the pagination controls to not show and the summary should say 'Showing 1-6 of 21 items'
> 
> Screen Shot 2024-04-02 at 1 50 30 PM

Hmm yes I think we should look into this separately on Grommet.

**Pull request merged**